ts/mcl.ts

```diff
@@ -28,132 +39,120 @@ export function setDomainHex(domain: string) {
     return g1;
 }
 
-export function g2() {
+export function g2(): mclG2 {
     const g2 = new mcl.G2();
     g2.setStr(
         "1 0x1800deef121f1e76426a00665e5c4479674322d4f75edadd46debd5cd992f6ed 0x198e9393920d483a7260bfb731fb5d25f1aa493335a9e71297e485b7aef312c2 0x12c85ea5db8c6deb4aab71808dcb408fe3d1e7690c43d37b4ce6cc0166fa7daa 0x090689d0585ff075ec9e99ad690c3395bc4b313370b38ef355acdadcd122975b"
     );
     return g2;
 }
 
-export function g1ToBig(p: mclG1) {
+export function g1ToHex(p: mclG1): solG1 {
     p.normalize();
-    const x = toBig(mclToHex(p.getX()));
-    const y = toBig(mclToHex(p.getY()));
+    const x = hexlify(toBigEndian(p.getX()));
+    const y = hexlify(toBigEndian(p.getY()));
     return [x, y];
 }
 
-export function g1ToHex(p: mclG1) {
+export function g2ToHex(p: mclG2): solG2 {
     p.normalize();
-    const x = mclToHex(p.getX());
-    const y = mclToHex(p.getY());
-    return [x, y];
-}
-
-export function g2ToBig(p: mclG2) {
-    const x = mclToHex(p.getX(), false);
-    const y = mclToHex(p.getY(), false);
-    return [
-        toBig("0x" + x.slice(64)),
-        toBig("0x" + x.slice(0, 64)),
-        toBig("0x" + y.slice(64)),
-        toBig("0x" + y.slice(0, 64))
-    ];
+    const x = toBigEndian(p.getX());
+    const x0 = hexlify(x.slice(32));
+    const x1 = hexlify(x.slice(0, 32));
+    const y = toBigEndian(p.getY());
+    const y0 = hexlify(y.slice(32));
+    const y1 = hexlify(y.slice(0, 32));
+    return [x0, x1, y0, y1];
 }
 
-export function g2ToHex(p: mclG2) {
-    p.normalize();
-    const x = mclToHex(p.getX(), false);
-    const y = mclToHex(p.getY(), false);
-    return [
-        "0x" + x.slice(64),
-        "0x" + x.slice(0, 64),
-        "0x" + y.slice(64),
-        "0x" + y.slice(0, 64)
-    ];
-}
-
-export function newKeyPair() {
+export function newKeyPair(): keyPair {
     const secret = randFr();
-    const pubkey = mcl.mul(g2(), secret);
-    pubkey.normalize();
+    const mclPubkey = mcl.mul(g2(), secret);
+    mclPubkey.normalize();
+    const pubkey = g2ToHex(mclPubkey);
     return { pubkey, secret };
 }
 
-export function sign(message: string, secret: mclFR) {
-    const M = hashToPoint(message);
-    const signature = mcl.mul(M, secret);
+export function sign(
+    message: string,
+    secret: SecretKey
+): {
+    signature: Signature;
+    M: Message;
+} {
+    const messagePoint = hashToPoint(message);
+    const signature = mcl.mul(messagePoint, secret);
     signature.normalize();
+    const M = g1ToHex(messagePoint);
     return { signature, M };
 }
 
-export function aggreagate(acc: mclG1 | mclG2, other: mclG1 | mclG2) {
-    const _acc = mcl.add(acc, other);
-    _acc.normalize();
-    return _acc;
+export function aggreagate(signatures: Signature[]): solG1 {
+    let aggregated = new mcl.G1();
+    for (const sig of signatures) {
+        aggregated = mcl.add(aggregated, sig);
+    }
+    aggregated.normalize();
+    return g1ToHex(aggregated);
 }
```

Comment:
Change the aggregate to aggregate only the signatures. We are not aggregating anything else.